openeo_driver/integrations/s3/credentials.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import os
+from typing import TypedDict, Optional
+
+_log = logging.getLogger(__name__)
+
+
+class Boto3CredentialsTypeDef(TypedDict):
+    aws_access_key_id: str
+    aws_secret_access_key: str
+
+
+def get_credentials(region_name: str, provider_name: str) -> Boto3CredentialsTypeDef:
+    """
+    Credential resolving should always go from most specific to least specific. So let's say we are checking region
+    waw3-1 of the cloudferro cloud we must check in order:
+    - WAW3_1_ACCESS_KEY_ID && WAW3_1_SECRET_ACCESS_KEY
+    - CF_ACCESS_KEY_ID && CF_SECRET_ACCESS_KEY
+    - SWIFT_ACCESS_KEY_ID && SWIFT_SECRET_ACCESS_KEY  # for backwards compatibility
+    - AWS_ACCESS_KEY_ID && AWS_SECRET_ACCESS_KEY
+    """
+    for prefix in [f"{region_name}", f"{provider_name}", "SWIFT", "AWS"]:
+        credential_candidate = get_credential_for_prefix(prefix)
+        if credential_candidate is not None:
+            return credential_candidate
+    raise EnvironmentError(f"Cannot get credentials for {region_name}")
+
+
+def _sanitize_env_name(env_name: str) -> str:
+    return env_name.upper().replace("-", "_")
```

Comment:
why not go full regex to be sure? Something like  (untested)

```suggestion
    re.sub("[^A-Z0-9]+", "_", env_name.upper())
```